server/src/internalClusterTest/java/org/opensearch/indices/IndicesRequestCacheIT.java

```diff
@@ -677,14 +689,324 @@ public void testCacheWithInvalidation() throws Exception {
         assertCacheState(client, "index", 1, 2);
     }
 
+    // when staleness threshold is low, it should clean the cache
+    public void testStaleKeysCleanup_LowThresholdCleansUpCache() throws Exception {
+        Instant start = Instant.now();
+        long thresholdInMillis = 1_500;
+        String node = internalCluster().startNode(
+            Settings.builder()
+                .put(IndicesRequestCache.SETTING_INDICES_REQUEST_CACHE_STALENESS_THRESHOLD_SETTING, 0.10)
+                .put(IndicesRequestCache.SETTING_INDICES_REQUEST_CACHE_CLEAN_INTERVAL_SETTING, TimeValue.timeValueMillis(thresholdInMillis))
+        );
+        String index = "index";
+        Client client = client(node);
+        setupIndex(client, index);
+
+        // create first cache entry
+        createCacheEntry(client, index, "hello");
+        assertCacheState(client, index, 0, 1);
+        long expectedFirstCachedItemEntrySize = getRequestCacheStats(client, index).getMemorySizeInBytes();
+        assertTrue(expectedFirstCachedItemEntrySize > 0);
+
+        // create second cache entry
+        createCacheEntry(client, index, "there");
+        assertCacheState(client, "index", 0, 2);
+        assertEquals(expectedFirstCachedItemEntrySize * 2, getRequestCacheStats(client, "index").getMemorySizeInBytes());
+
+        // force refresh so that it creates 2 stale keys in the cache for the cache cleaner to pick up.
+        flushAndRefresh("index");
+        client().prepareIndex("index").setId("1").setSource("k", "good bye");
+        ensureSearchable("index");
+
+        // create another entry
+        createCacheEntry(client, index, "hello1");
+        assertCacheState(client, "index", 0, 3);
+        long currentMemorySize = getRequestCacheStats(client, "index").getMemorySizeInBytes();
+        long expectedThirdCachedEntrySize = currentMemorySize - (expectedFirstCachedItemEntrySize * 2);
```

Comment:
This will be flaky, as by this time the cache cleanup may have already been triggered.

I think you can instead do this:
1. Create 2 indices with 1 shard each
2. Keep the cleanup interval very low like 1 sec. 
3. Add 2 entries in the cache for index 1 and 1 entry in cache for index 2, verify that entries are present and hits/misses count
4. Perform flush/refresh on index2 after adding new document
5. Sleep and wait for the cleanup interval time.
6. Verify the cleanup has been done for the 1 entry of index2 and other 2 entries for index 1 still exists (since stale keys will be ~33%)

Reply:
I like this approach. thanks !